systemd: 258.3 -> 259

arianvp opened the pull request:
## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [x] x86_64-linux (glibc)
  - [x] x86_64-linux (musl)
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [ ] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [x] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [ ] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

stelcodes commented:
Please backport this if possible to fix the nasty systemd-boot bug (#461676) currently affecting stable. It led to bad bootloader entries on my machine that ended up in a rescue shell. The supposed fix (https://github.com/systemd/systemd/pull/39715) was released in v259. Thanks so much to everyone working on this 👍

LordGrimmauld commented:
https://git.grimmauld.de/Grimmauld/grimm-musl/commit/064dd8afcfc41c2083c0b8842b60e99bf6282fb5

<img width="1885" height="477" alt="image" src="https://github.com/user-attachments/assets/72a7454c-6d42-42fd-b08c-a13e2ca6eb52" />

#473403, #453557, #472008, #473068 on musl nixos and current staging, with a couple other fixes

arianvp committed:
systemd: 258.3 -> 259

Had to revert https://github.com/NixOS/nixpkgs/pull/404902 as it gives
linker errors. And I don't think the systemd tests work on NixOS anyway.

LordGrimmauld committed:
systemd: drop all musl patches

rhendric commented:
Per [NEWS](https://github.com/systemd/systemd/blob/51190631968f2a69acf5da3e3412b003805538f2/NEWS#L314), we should be able to drop the libcap dependency here.

arianvp committed:
systemd: drop libcap dependency

woohoo! one less circular dep to worry about

from NEWS:
systemd no longer links against libcap at all. The simple system call
wrappers and other APIs it provides have been reimplemented directly
in systemd, which reduced the codebase and the dependency tree.

WilliButz reviewed (approved):
rebased the commits locally on a recent staging rev, built a couple of images and ran a some of the in-tree + some downstream integration tests. LGTM 👍
